src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -72,27 +72,31 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     webapp_def = Site(location=location, site_config=site_config, server_farm_id=plan_info.id, tags=tags)
     helper = _StackRuntimeHelper(client, linux=is_linux)
 
-    if is_linux:
-        if not validate_linux_create_options(runtime, deployment_container_image_name,
-                                             multicontainer_config_type, multicontainer_config_file):
+    if is_linux or plan_info.is_xenon:
+        if not validate_container_app_create_options(runtime, deployment_container_image_name,
+                                                     multicontainer_config_type, multicontainer_config_file):
             raise CLIError("usage error: --runtime | --deployment-container-image-name |"
                            " --multicontainer-config-type TYPE --multicontainer-config-file FILE")
-        if startup_file:
+        if is_linux and startup_file:
             site_config.app_command_line = startup_file
 
-        if runtime:
+        if is_linux and runtime:
```

Comment:
why are you adding  check for is_linux here - currently we allow passing a runtime for windows webapps that not containers. Ex. https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az-webapp-create

Reply:
Hi @panchagnula . Since we changed the larger if ('is_linux' to 'is_linux or plan_info.is_xenon') we need to be able to distinguish between linux or xenon and handle them accordingly.

The case you mention is still being handled in the line [101:] `elif runtime:  # windows webapp with runtime specified`